controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -658,3 +692,13 @@ func (r *StorageConsumerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&rookCephv1.CephClient{}).
 		Complete(r)
 }
+
+func generateHashForCephClient(storageConsumerName, claimID, cephUserType string) string {
```

Comment:
Make this a method of `cephClientFormat`?

Reply:
could you explain the use case here or do you have examples of how we use it?

Reply:
instead of adding or passing more parameters make this a method it will help in future also

```go
func (c *cephClientFormat)generateHashForCephClient() string {
cephClient, err := json.Marshal(c)
if err != nil {
	panic("failed to marshal")
 }
return hasvalue?
}
```

Reply:
going with this approach https://github.com/red-hat-storage/ocs-operator/pull/1564#discussion_r815784288